core/src/OpenACC/Kokkos_OpenACC_ParallelScan_Range.hpp

```diff
@@ -19,8 +19,27 @@
 
 #include <OpenACC/Kokkos_OpenACC.hpp>
 #include <OpenACC/Kokkos_OpenACC_FunctorAdapter.hpp>
+#include <OpenACC/Kokkos_OpenACC_Macros.hpp>
 #include <Kokkos_Parallel.hpp>
 
+// Clacc uses an alternative implementation to work around not-yet-implemented
+// OpenACC features: Clacc does not fully support private clauses for
+// gang-private variables, and the alternative implementation allocates
+// the gang-private arrays on GPU global memory using array expansion,
+// instead of using the private clause.
+/* clang-format off */
+#ifdef KOKKOS_COMPILER_CLANG
+#define ELEMENT_VALUES_SIZE (n_chunks * 2 * chunk_size)
+#define ACCESS_ELEMENTS(THREADID) \
+  element_values[team_id * 2 * chunk_size + THREADID]
```

Comment:
I think that the current approach is better than using an unmanaged view in terms of the number of code changes between two implementations (one implementation uses the element_values as a gang-private array and the other uses it as a gang-shared array).

Reply:
You need to prefix all macros with `KOKKOS_IMPL_[ACC_]`.

What prevents you to use a regular variable instead of a macro for the number of elements in the array?

~`ELEMENT_VALUES_SIZE`~
```C++
#ifdef KOKKOS_COMPILER_CLANG
int const num_elements = n_chunks * 2 * chunk_size;
#else
int const num_elements = 2 * chunk_size;
#endif
```

Reply:
Prefixed all macros with `KOKKOS_IMPL_ACC_`.
Removed ELEMENT_VALUES_SIZE.